starlette/middleware/base.py

```diff
@@ -26,6 +88,8 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             await self.app(scope, receive, send)
             return
 
+        request = _CachedRequest(scope, receive)
```

Comment:
This doesn't need to be here, does it? Can it be in the same place it was instantiated before, or am I missing something?

Reply:
The next line references this. It, maybe, can be moved to where it was before but at the very least it will need a new variable name  like`outer_request` to differentiate it from the `request: Request` on line 95. It makes more sense to just move it up here, there is no harm in that.